File: lib/src/model/kakao_map_position.dart
part of '../../flutter_kakao_maps.dart';

class KakaoMapPosition {
  final KakaoMapAlignment alignment;
  final double x;
  final double y;

  const KakaoMapPosition({
    required this.alignment,
    required this.x,
    required this.y,
  });

  KakaoMapPosition copyWith({
    KakaoMapAlignment? alignment,
    double? x,
    double? y,
  }) =>
      KakaoMapPosition(
        alignment: alignment ?? this.alignment,
        x: x ?? this.x,
        y: y ?? this.y,
      );
}

extension KakaoMapPositionExtension on KakaoMapPosition {
  Map<String, dynamic> toMap() {
    return {
      "alignment": alignment.toString(),
      "x": x,
      "y": y,
    };
  }
}

File: ios/flutter_kakao_maps/Sources/flutter_kakao_maps/model/KakaoMapPosition.swift
import Foundation

struct KakaoMapPosition {
    var alignment: KakaoMapAlignment
    var x: Double
    var y: Double
}

extension NSDictionary {
    func toKakaoMapPosition() -> KakaoMapPosition {
        return KakaoMapPosition(alignment: KakaoMapAlignment(rawValue: self["alignment"] as! String)!, x: self["x"] as! Double, y: self["y"] as! Double)
    }
}
